samples/scriptable/line.html

```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE html>
+<html lang="en-US">
+<head>
+	<meta charset="utf-8">
+	<meta http-equiv="X-UA-Compatible" content="IE=Edge">
+	<meta name="viewport" content="width=device-width, initial-scale=1">
+	<title>Scriptable > Line | Chart.js sample</title>
+	<link rel="stylesheet" type="text/css" href="../style.css">
+	<script src="../../dist/Chart.min.js"></script>
+	<script src="../utils.js"></script>
+</head>
+<body>
+	<div class="content">
+		<div class="wrapper"><canvas id="chart-0"></canvas></div>
+		<div class="toolbar">
+			<button onclick="randomize(this)">Randomize</button>
+			<button onclick="addDataset(this)">Add Dataset</button>
+			<button onclick="removeDataset(this)">Remove Dataset</button>
+		</div>
+	</div>
+
+	<script>
+		var DATA_COUNT = 12;
+
+		var utils = Samples.utils;
+
+		utils.srand(110);
+
+		function alternatePointStyles(ctx) {
+			var index = ctx.dataIndex;
+			return index % 2 === 0 ? 'circle' : 'rect';
+		}
+
+		function makeHalfAsOpaque(ctx) {
+			var c = ctx.dataset.backgroundColor;
+			return utils.transparentize(c);
+		}
+
+		function adjustRadiusBasedOnData(ctx) {
+			var v = ctx.dataset.data[ctx.dataIndex];
+			return v < 10 ? 5
+				: v < 25 ? 7
+				: v < 50 ? 9
+				: v < 75 ? 11
+				: 15;
+		}
+
+		function generateData() {
+			return utils.numbers({
+				count: DATA_COUNT,
+				min: 0,
+				max: 100
+			});
+		}
+
+		var data = {
+			labels: utils.months({count: DATA_COUNT}),
+			datasets: [{
+				data: generateData(),
+				backgroundColor: '#4dc9f6',
+				borderColor: '#4dc9f6',
```

Comment:
I was about to ask to make these two options scriptable but realized it's not *yet* scriptable for the line element. Once implemented, we should be able to do something like (as part of a different PR):

```javascript
borderColor: function(context) {
    // make the line and point border color the 
    // same while different for each dataset.
    return utils.color(context.datasetIndex);
}
```

Reply:
@simonbrunel Maybe this is a consideration for radar chart as well? Adding scriptable line backgroundColor and borderColor?

Reply:
Yes, you right, line and radar controllers are very similar (currently trying to refactor them).